frontend/src/dispatcher/js-dispatcher.ts

```diff
@@ -30,26 +31,26 @@ export function createJsDispatcher() {
 			return;
 		}
 
+		// TODO: Provide an explanatory comment here
+		const isJsMessageConstructor = (fn: typeof messageConstructor): fn is typeof JsMessage => "jsMessageMarker" in fn;
+		const messageIsConstructor = isJsMessageConstructor(messageConstructor);
+
 		// Messages with non-empty data are provided by wasm-bindgen as an object with one key as the message name, like: { NameOfThisMessage: { ... } }
 		// Messages with empty data are provided by wasm-bindgen as a string with the message name, like: "NameOfThisMessage"
+		// Here we extract the payload object or use an empty object depending on the situation.
 		const unwrappedMessageData = messageData[messageType] || {};
 
-		const isJsMessageConstructor = (fn: typeof messageConstructor): fn is typeof JsMessage => "jsMessageMarker" in fn;
-		let message: JsMessage;
-		if (isJsMessageConstructor(messageConstructor)) {
-			message = plainToInstance(messageConstructor, unwrappedMessageData);
-		} else {
-			message = messageConstructor(unwrappedMessageData, wasm, instance);
-		}
+		// TODO: Provide an explanatory comment here
+		const message = messageIsConstructor ? plainToInstance(messageConstructor, unwrappedMessageData) : messageConstructor(unwrappedMessageData, wasm, instance);
```

Comment:
Is it possible to put this on multiple lines?

Reply:
Nope. Autoformatter will put it back.